tests/setup/dualstack/doc_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package setupconfig
+
+import (
+	"testing"
+
+	"istio.io/istio.io/pkg/test/istioio"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+func TestMain(m *testing.M) {
+	// nolint: staticcheck
+	framework.
+		NewSuite(m).
+		Setup(istio.Setup(nil, setupConfig)).
+		Run()
+}
+
+func TestDocs(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(istioio.NewTestDocsFunc("dualstack"))
```

Comment:
This requires a different kind cluster. I was not sure if this is related. so can we get this done without the need of a separate profile? https://github.com/istio/test-infra/pull/5106

Reply:
Ideally yes, we should be able to test other docs page as well on the dual-stack cluster eventually.. currently do we have such options for running docs against ipv6 cluster?

Reply:
if we need a way to run all the docs pages against dual-stack cluster, we need a better way than profile=dualstack.
a way to specify, profile=none, cluster=ipv4,ipv6,dual-stack so that the test runs in all three jobs. But I don't think that is how the tests are currently.

Reply:
LGTM. Will see if @frankbu has any other insights on this, noting that this test needs a different cluster than the normal profile-* tests (which is similar to multi cluster having a different cluster setup). Besides the cluster difference, this does need some additional setup config for Istio (which again is similar to multicluster). Currently, this allows for additional tests to be written in the future using the dual-stack cluster and config.